packages/kiota_abstractions/lib/src/request_configuration.dart

```diff
@@ -1,13 +1,12 @@
 part of '../kiota_abstractions.dart';
 
 /// Request configuration type for [BaseRequestBuilder]s.
-class RequestConfiguration {
+class RequestConfiguration<T extends AbstractQueryParameters> {
   /// The HTTP headers of the request.
   HttpHeaders headers = HttpHeaders();
 
   /// The request options.
   List<RequestOption> options = [];
 
-  /// The request query parameters.
-  Map<String, dynamic> queryParameters = {};
+  late T queryParameters;
```

Comment:
I don't this is a good design. It should only be marked `late` if you can guarantee that this value will be set before being accessed.

I'd opt to make this nullable instead.

Reply:
Joanne fixed this. The late keyword is gone.